tests/artifacts_test.go

```diff
@@ -0,0 +1,207 @@
+		require.NoError(t, err, "Failed to create pipeline run")
+		responseData, err = io.ReadAll(response.Body)
+		require.NoError(t, err, "Failed to read run response data")
+		require.Equal(t, http.StatusOK, response.StatusCode, "Unexpected HTTP status code")
+		log.Println("Pipeline run created successfully.")
+
+		err = TestUtil.WaitForPipelineRunCompletion(t, suite.Clientmgr.httpClient, APIServerURL)
+		require.NoError(t, err, "Pipeline run did not complete successfully")
+
+		// Fetch artifacts from API
+		artifactsUrl := fmt.Sprintf("%s/apis/v2beta1/artifacts?namespace=%s", APIServerURL, suite.DSPANamespace)
+		log.Printf("Fetching artifacts from URL: %s", artifactsUrl)
+		response, err = suite.Clientmgr.httpClient.Get(artifactsUrl)
+		require.NoError(t, err, "Failed to fetch artifacts")
+		responseData, err = io.ReadAll(response.Body)
+		require.NoError(t, err, "Failed to read artifacts response data")
+		assert.Equal(t, http.StatusOK, response.StatusCode, "Unexpected HTTP status code")
+
+		// Parse the artifact list
+		var responseArtifactsData struct {
+			Artifacts []ResponseArtifact `json:"artifacts"`
+		}
+		err = json.Unmarshal(responseData, &responseArtifactsData)
+		require.NoError(t, err, "Failed to parse artifacts response JSON")
+
+		hasDownloadError := false
+
+		for _, artifact := range responseArtifactsData.Artifacts {
+			if artifact.ArtifactType != "system.Model" {
+				continue
+			}
+
+			// Fetch download URL
+			artifactsByIdUrl := fmt.Sprintf("%s/apis/v2beta1/artifacts/%s?view=DOWNLOAD", APIServerURL, artifact.ArtifactID)
+			log.Printf("Fetching download URL for artifact from: %s", artifactsByIdUrl)
+			response, err = suite.Clientmgr.httpClient.Get(artifactsByIdUrl)
+			require.NoError(t, err, "Failed to fetch download URL")
+			responseData, err = io.ReadAll(response.Body)
+			require.NoError(t, err, "Failed to read download URL response data")
+			assert.Equal(t, http.StatusOK, response.StatusCode, "Unexpected HTTP status code")
+
+			var artifactWithDownload ResponseArtifact
+			err = json.Unmarshal(responseData, &artifactWithDownload)
+			require.NoError(t, err, "Failed to parse download URL response JSON")
+
+			// Modify the URL for local port-forwarding
+			parsedURL, err := url.Parse(artifactWithDownload.DownloadUrl)
+			require.NoError(t, err, "Failed to parse artifact download URL")
+
+			originalHost := parsedURL.Host
+			parsedURL.Host = "127.0.0.1:9000"
+
+			log.Printf("Trying download URL: %s", parsedURL.String())
+
+			// Create and send the request with correct Host header
+			req, err := http.NewRequest("GET", parsedURL.String(), nil)
+			require.NoError(t, err, "Failed to create request")
+
+			req.Host = originalHost
+
+			// Create a custom HTTP client that ignores TLS verification
+			httpClient := &http.Client{
+				Transport: &http.Transport{
+					TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, //Fix TLS verification error
```

Comment:
I see your to do comment here. I'm okay with just leaving it unverified and if you are too, we can just remove the comment.

Reply:
That was actually an insight into why we used `InsecureSkipVerify: true` but yes, it's causing confusion hence took it out :D